File: AndroidStudioStudy/2-TheLightOfAdvanced/Chapter6-DesignPatterns/DesignPatternsDemo/app/src/main/java/com/bo/designpatternsdemo/MainActivity.java
package com.bo.designpatternsdemo;

import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.FragmentManager;

import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import com.bo.designpatternsdemo.databinding.ActivityMainBinding;

public class MainActivity extends BaseActivity {
    ActivityMainBinding activityMainBinding;
    private Button btSimpleFactory;
    private Button btFactoryMethod;
    private Button btBuilder;
    private Button btProxy;
    private Button btDecorator;
    private Button btFacade;
    private Button btFlyweight;
    private Button btStrategy;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        activityMainBinding=ActivityMainBinding.inflate(getLayoutInflater());
        setContentView(activityMainBinding.getRoot());
    }

    @Override
    protected void initViews() {
        super.initViews();

        btSimpleFactory = activityMainBinding.btSimpleFactory;
        btFactoryMethod = activityMainBinding.btFactoryMethod;
        btBuilder = activityMainBinding.btBuilder;
        btProxy = activityMainBinding.btProxy;
        btDecorator = activityMainBinding.btDecorator;
        btFacade = activityMainBinding.btFacade;
        btFlyweight = activityMainBinding.btFlyweight;
        btStrategy = activityMainBinding.btStrategy;
    }

    @Override
    protected void initListeners() {
        super.initListeners();

        btSimpleFactory.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new SimpleFactory().createComputer(SimpleFactory.LENOVO).start();
            }
        });

        btFactoryMethod.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new FactoryMethod().createComputer();
            }
        });

        btBuilder.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                BuilderTest.start();
            }
        });

        btProxy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                //new ProxyTest().buy();
                new ProxyTest().buyDynamic();
            }
        });

        btDecorator.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new DecoratorTest().use();
            }
        });

        btFacade.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new FacadeTest().test();
            }
        });

        btFlyweight.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new FlyweightTest().test();
            }
        });

        btStrategy.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                new StrategyTest().test();
            }
        });
    }
}

File: Internship-Android-Markdown/view-learn/AnimatorDemo/viewpropertyanimatordemo/src/main/java/com/example/viewpropertyanimatordemo/MainActivity.kt
package com.example.viewpropertyanimatordemo

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.view.View
import com.example.viewpropertyanimatordemo.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity(), View.OnClickListener {
    private lateinit var binding: ActivityMainBinding

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        binding= ActivityMainBinding.inflate(layoutInflater)
        setContentView(binding.root)

        binding.btnViewPropertyAnimator.setOnClickListener(this)
    }

    override fun onClick(v: View?) {
        when(v!!.id){
            binding.btnViewPropertyAnimator.id->{
                binding.btnViewPropertyAnimator.animate().apply {
                    rotationX(360f)
                    rotationY(360f)
                    rotation(360f)
                    alpha(0f).alpha(1f)
                    x(500f).x(200f)
                    y(1200f).y(200f)
                    duration = 5000
                    start()

                }
            }
        }
    }
}